server/cli/src/load_test.rs

```diff
@@ -218,93 +367,250 @@ impl LoadTest {
-                    interval_seconds: 600,
-                    batch_size: BatchSize {
-                        remote_pull: 512,
-                        remote_push: 512,
-                        central_pull: 512,
-                    },
-                }),
-                logging: None,
-                backup: None,
-                mail: None,
-            };
-
-            std::fs::write(&config_file_path, serde_yml::to_string(&config)?)?;
-            println!("Created config file {:?}", config_file_path);
+        for test_site in &test_sites {
+            std::fs::write(
+                &test_site.config_file_path.clone(),
+                serde_yml::to_string(&test_site.settings.clone())?,
+            )?;
         }
 
         // Start each remote OMS instance
         println!("Starting remote OMS instances...");
-        let mut child_processes = Vec::new();
-
-        let output = Command::new("pwd").output()?;
-        std::io::stdout().write_all(&output.stdout)?;
-
-        for site_n_store in site_n_stores.iter() {
-            let config_file_path = self
-                .output_dir
-                .join(format!("site_{}_config.yaml", site_n_store.site.site_id));
-
-            let child = Command::new("cargo")
-                .arg("run")
-                .arg("--")
-                .arg("--config-path")
-                .arg(&config_file_path)
-                .spawn()?;
-
-            child_processes.push(child);
-
-            // Give a little time for the instance to start before launching the next one
-            sleep(Duration::from_millis(200)).await
+        let mut handles = Vec::new();
+        let duration = self.duration as u64;
+        let num_lines = self.lines;
+
+        for test_site in test_sites {
+            let dir = self.output_dir.clone();
+            let item_ids_copy = item_ids.clone();
+            let handle = tokio::spawn(async move {
+                let log = std::fs::File::create(
+                    dir.join(format!("site_{}_output.log", test_site.site.site_id)),
+                )
+                .unwrap();
+                let mut child = match Command::new("cargo") // TODO: be better to run prod binary instead
+                    .arg("run")
+                    .arg("--")
+                    .arg("--config-path")
+                    .arg(&test_site.config_file_path)
+                    .stdout(log.try_clone().unwrap())
+                    .stderr(log)
+                    .env("RUST_LOG", "none")
+                    .kill_on_drop(true)
+                    .spawn()
+                {
+                    Ok(child) => child,
+                    Err(e) => {
+                        println!("Failed to spawn process: {}", e);
+                        return;
+                    }
+                };
+                sleep(Duration::from_secs(10)).await; // Let db get created, migrated and initialisation started
```

Comment:
ah I had this scaled to amount of sites at one point, it may not be reliable at 10s if you start say 1000 sites. Though you're likely pretty crazy to do that on one machine anyway, unless it's some 128core beast or something.

_Ideally_, I'd do something like capture stdout in process here and continue when the server states that it's finished migration and started initialisation.

Reply:
Agree, think that would be better... plus a log here, something like "remote sites started, beginning load test..."?